### DISCUSSION TITLE
How to interpret the number of anomaly counts?

### DISCUSSION THREAD

**Discussion by MengLu-flw** [LohseLab/gIMble#123] [Q&A] [answered]
Hi :^)

I really like the ‘gimble query’ function, which gives a very clean summary of the ‘gimble optimize’ outputs. 

However, I was wondering how I shall interpret the number of anomaly counts (e.g., does a large number of anomaly counts mean that a model fits badly, despite it shows a high lnCL value compared to the other models?)

Detailed descriptions:
I fitted my data to all five available models in gIMble (DIV, IM_AB, IM_BA, MIG_AB, MIG_BA), and ran them with the code like:
`gimble optimize -g CRS2 -e 39 [I change this random seed number for each run] -i 10000`

For all IM models (regardless of which dataset I used), they tend to give high anomaly counts (around 5000 anomalies out of 10000) – I feel that cannot be a good thing (?)

Is there any way that I could improve this abnormal behaviour?

Very looking forward to your insightful advice, and many thanks in advance!

Gratefully,
Meng

**Comment by GertjanBisschop** [comment, marked as answer]
Hi @MengLu-flw, 
Thanks for flagging this. We should indeed provide a bit more explanation on what the reported number of anomalies mean. 

The reported anomalies have to do with numerical stability issues that are sometimes encountered when computing the probabilities of seeing particular mutation configurations given a certain set of parameter combinations. These probabilities are computed using the `agemo` library. A little bit of work is required on that front to improve the numerical stability (see [this issue)](https://github.com/LohseLab/agemo/issues/6). There is nothing you can do really to improve this, apart perhaps from using a different set of random seeds to explore parameter space. However, given the large number of anomalies, it seems very likely that in your case the optimal region of parameter space is particularly prone to numerical instability. And that different seeds would induce the same behaviour.
Luckily enough, the estimates that are not affected by numerical instability should be accurate and should thus not affect your ability to interpret the outcome of these optimisation run.

**Reply by MengLu-flw** [reply]
Hi @GertjanBisschop,

Thank you so much for your prompt reply and great help!

Best regards,
Meng
